system/jlib/jstats.cpp

```diff
@@ -2190,49 +2350,59 @@ IStatisticCollection * createStatisticCollection(MemoryBuffer & in)
     return deserializeCollection(NULL, in, version);
 }
 
+IStatisticCollection * createStatisticCollection(IStatisticCollection * parent, const StatsScopeId & scopeId)
+{
+    return new CStatisticCollection(parent, scopeId);
+}
+
+IStatisticCollection * createRootStatisticCollection(StatisticCreatorType creatorType, const char * creator, const StatsScopeId & rootScope, const StatsScopeId & graphScopeId, IStatisticCollection * sgCollection)
+{
+    //creator unused at the moment.
+    return new CRootStatisticCollection(creatorType, creator, rootScope, graphScopeId, sgCollection);
+}
 
 //--------------------------------------------------------------------------------------------------------------------
 
 class StatisticGatherer : implements CInterfaceOf<IStatisticGatherer>
 {
 public:
-    StatisticGatherer(CStatisticCollection * scope) : rootScope(scope)
+    StatisticGatherer(IStatisticCollection * scope) : rootScope(scope)
     {
         scopes.append(*scope);
     }
     virtual void beginScope(const StatsScopeId & id) override
     {
-        CStatisticCollection & tos = scopes.tos();
+        IStatisticCollection & tos = scopes.tos();
         scopes.append(*tos.ensureSubScope(id, true));
     }
     virtual void beginActivityScope(unsigned id) override
     {
         StatsScopeId scopeId(SSTactivity, id);
-        CStatisticCollection & tos = scopes.tos();
+        IStatisticCollection & tos = scopes.tos();
         scopes.append(*tos.ensureSubScope(scopeId, false));
     }
     virtual void beginSubGraphScope(unsigned id) override
     {
         StatsScopeId scopeId(SSTsubgraph, id);
-        CStatisticCollection & tos = scopes.tos();
+        IStatisticCollection & tos = scopes.tos();
         scopes.append(*tos.ensureSubScope(scopeId, true));
     }
     virtual void beginEdgeScope(unsigned id, unsigned oid) override
     {
         StatsScopeId scopeId(SSTedge, id, oid);
-        CStatisticCollection & tos = scopes.tos();
+        IStatisticCollection & tos = scopes.tos();
         scopes.append(*tos.ensureSubScope(scopeId, false));
     }
     virtual void beginChildGraphScope(unsigned id) override
     {
         StatsScopeId scopeId(SSTchildgraph, id);
-        CStatisticCollection & tos = scopes.tos();
+        IStatisticCollection & tos = scopes.tos();
         scopes.append(*tos.ensureSubScope(scopeId, true));
     }
     virtual void beginChannelScope(unsigned id) override
     {
         StatsScopeId scopeId(SSTchannel, id);
-        CStatisticCollection & tos = scopes.tos();
+        IStatisticCollection & tos = scopes.tos();
```

Comment:
Still lots of unrequired changes from CStatisticCollection to IStatisticCollection throught the file that complicate the compare.